bundles/core/src/main/java/com/adobe/cq/wcm/core/components/commons/link/Link.java

```diff
@@ -74,13 +72,24 @@ default String getURL() {
     }
 
     /**
-     * The processed URL, in case any extra processing can be done.
+     * The mapped URL, which supports mapping and vanity path.
      * This usually is resource resolver mapping.
      *
-     * @return Process link URL or {@code null} if link is invalid or no processing can be done
+     * @return Mapped link URL or {@code null} if link is invalid or no processing can be done
      */
     @Nullable
-    default String getProcessedURL() {
+    default String getMappedURL() {
+        return null;
+    }
+
+
+    /**
+     * The externalized URL which also contains the scheme and host information.
+     * This is usually created with a {@link com.day.cq.commons.Externalizer} service
+     * @return Full link URL or {@code null} if link is invalid or can't be externalized.
```

Comment:
There's a disconnect here in documentation. On one hand `Link#getExternalizedURL` claims to return `null` if link can't be externalized, on the other hand the `PathProcessor#externalizePath` is annotated to return `@NotNull` (and the default implementation will return the original/nonexternalized path in case of errors).

Reply:
If the url is null and the link is invalid also the Link#getExteralizedURL will return null now.